modules/tinymce/src/themes/silver/main/ts/ui/menus/item/build/ColorSwatchItem.ts

```diff
@@ -53,3 +53,13 @@ export const renderColorSwatchItem = (spec: Menu.FancyMenuItem, backstage: UiFac
     ]
   };
 };
+
+const getColorItems = (spec: Menu.FancyMenuItem, backstage: UiFactoryBackstage): Menu.ChoiceMenuItemSpec[] => {
+  if (Type.isArray(spec.initData.colors)) {
+    return spec.initData.colors.concat(ColorSwatch.getAdditionalColors(false));
```

Comment:
This needs to respect the `ignoreCustomColors` setting, not always pass `false`. So this logic should be something like this instead:
```
const useCustomColors = spec.initData.ignoreCustomColors !== true;
if (Type.isArray(spec.initData.colors)) {
  return spec.initData.colors.concat(ColorSwatch.getAdditionalColors(useCustomColors));
} else {
  return ColorSwatch.getColors(backstage.colorinput.getColors(), useCustomColors && backstage.colorinput.hasCustomColors());
}
```

- [ ] Fix this logic so it properly ignores as per the initial data provided.

Reply:
I should add you'll need to set the `ignoreCustomColors` in your new buttons. I'd also like to see this renamed to something like `allowCustomColors` instead as that's more accurate to what it's doing.

Reply:
I don't think it'd be a good idea to rename it to `allowCustomColors` as that'd change it from us actively disallowing custom colors to having to actively allow them, which would break how things currently work.

Reply:
I discussed this with the team and I'm not sure if there was a misunderstanding of what I wrote earlier, but the team would prefer this to be called `allowCustomColors`. I also need to clarify that the expected behaviour here should be that custom colors are included by default from the specs perspective (doing otherwise would be a breaking change).

The reason this name is preferred is it's avoiding doing a double negative that's currently going on here, as by default it's enabled and if you want to disable it you just set `allowCustomColors: false` in your spec.

Reply:
To clarify more, that also means the start of this function would end up as:
```
const useCustomColors = spec.initData.allowCustomColors !== false  && backstage.colorinput.hasCustomColors();
```

Reply:
So then we'd have a property that allows custom colors if set to true, but also allows them if not set? That seems somewhat confusing and strange to deal with. I'd assume that with a name like `allowCustomColors` if it isn't set to true we wouldn't allow custom colors?